eng/cake/dotnet.cake

```diff
@@ -169,12 +173,22 @@ Task("dotnet-templates")
                     template.Value(projectName);
 
                 // Enable Tizen
-                ReplaceTextInFiles($"{projectName}/*.csproj",
-                    "<!-- <TargetFrameworks>",
-                    "<TargetFrameworks>");
-                ReplaceTextInFiles($"{projectName}/*.csproj",
-                    "</TargetFrameworks> -->",
-                    "</TargetFrameworks>");
+                if (!TestTFM.StartsWith("net6")) // Tizen does not yet support net6 on the net7 SDK: https://github.com/Samsung/Tizen.NET/issues/182
+                {
+                    ReplaceTextInFiles($"{projectName}/*.csproj",
+                        "<!-- <TargetFrameworks>",
+                        "<TargetFrameworks>");
+                    ReplaceTextInFiles($"{projectName}/*.csproj",
+                        "</TargetFrameworks> -->",
+                        "</TargetFrameworks>");
+                }
+                else
+                {
+                    // Mac Catalyst/iOS does not yet support net6 from the net7 SDK: https://github.com/xamarin/xamarin-macios/issues/15344
+                    ReplaceTextInFiles($"{projectName}/*.csproj",
+                        ";net6.0-ios;net6.0-maccatalyst",
+                        "");
+                }
```

Comment:
We have to disable building net6.0-ios and net6.0-maccatalyst in CI due to the fact that those SDKS do not yet support building .NET 6 apps from a .NET 7 SDK: https://github.com/xamarin/xamarin-macios/issues/15344